crates/virtio-queue/src/state.rs

```diff
@@ -311,6 +311,14 @@ impl QueueStateT for QueueState {
             .map_err(Error::GuestMemory)
     }
 
+    fn used_idx<M: GuestMemory>(&self, mem: &M, order: Ordering) -> Result<Wrapping<u16>, Error> {
+        let addr = self.used_ring.unchecked_add(2);
```

Comment:
I think I would use `checked_add` here for now because the queue's and the state's fields are public, so the user is allowed to configure a queue with invalid values.
I plan to update a bit the save/restore support for the queue, and we can reconsider the `(un)checked_*` story then.

Reply:
Cool